dask/array/routines.py

```diff
@@ -530,11 +530,18 @@ def round(a, decimals=0):
 
 @wraps(np.unique)
 def unique(x):
-    name = 'unique-' + x.name
-    dsk = {(name, i): (np.unique, key) for i, key in enumerate(x.__dask_keys__())}
-    parts = compute_as_if_collection(Array, sharedict.merge((name, dsk), x.dask),
-                                     list(dsk.keys()))
-    return np.unique(np.concatenate(parts))
+    x = x.ravel()
+
+    out = atop(np.unique, "i", x, "i", dtype=x.dtype)
+    out._chunks = tuple((np.nan,) * len(c) for c in out.chunks)
+
+    name = 'unique-aggregate-' + out.name
+    dsk = {(name, 0): (np.unique, (np.concatenate, out._keys()))}
```

Comment:
This use of `_keys` crept back in when I was rebasing apparently, but got fixed in PR ( https://github.com/dask/dask/pull/2833 ).